plone/app/iterate/relation.py

```diff
@@ -62,21 +68,29 @@ class CheckinCheckoutReferenceAdapter ( object ):
 
     forward refs on baseline get removed
     backward refs on baseline are kept by virtue of UID transferance
-    
+
     """
 
     implements( ICheckinCheckoutReference )
     adapts( IIterateAware )
-    
+
     storage_key = "coci.references"
 
     def __init__(self, context ):
         self.context = context
-    
-    def checkout( self, baseline, wc, refs, storage ):
+
+    def checkout(self, baseline, wc, refs, storage):
         for ref in refs:
-            wc.addReference( ref.targetUID, ref.relationship, referenceClass=ref.__class__ )
-            
+            try:
+                wc.addReference(ref.targetUID, ref.relationship,
+                                referenceClass=ref.__class__)
+            except ReferenceException:
+                logger.warn(
+                    'Reference exception when adding relation %r '
+                    'from new working copy %s to uid %s. Ignoring relation.',
+                    ref.relationship, '/'.join(wc.getPhysicalPath()),
+                    ref.targetUID)
+
```

Comment:
@mauritsvanrees you are only logging and ignoring the exception, so where's the magic removal? Some further down code that need not to be updated?